python/lsst/ctrl/oods/messageAttendant.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+
+from lsst.ctrl.oods.butlerAttendant import ButlerAttendant
+from lsst.ctrl.oods.imageData import ImageData
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MessageAttendant(ButlerAttendant):
+    """Processes files from S3 into the Butler.
+
+    Parameters
+    ----------
+    config: `dict`
+        configuration of this butler ingester
+    csc: `OodsCSC`
+        Observatory Operations Data Service Commandable SAL component
+    """
+
+    def rawexposure_info(self, data):
```

Comment:
A lot of these methods are (nearly) the same as the `FileAttendant`.  Maybe some can be moved into the base class?

Reply:
I went back and forth on moving everything to a base class and ultimately came to this for two reasons.  First, in order to refactor so it could be eliminated having the same functionality, the code would be less readable, especially for the methods used for callbacks.  Second, it's likely FileAttendant will go completely away when we fully transition to S3 storage.